pyro/infer/mcmc/hmc.py

```diff
@@ -18,17 +20,24 @@ class HMC(TraceKernel):
     Simple Hamiltonian Monte Carlo kernel, where ``step_size`` and ``num_steps``
     need to be explicitly specified by the user.
 
-    **Reference**
-    "MCMC Using Hamiltonian Dynamics", R Neal. `pdf <https://arxiv.org/pdf/1206.1901.pdf>`_
+    References
 
+    [1] `MCMC Using Hamiltonian Dynamics`,
+    Radford M. Neal
 
-    :param model: python callable containing pyro primitives.
-    :param float step_size: determines the size of a single step taken by the
+    :param model: Python callable containing pyro primitives.
+    :param float step_size: Determines the size of a single step taken by the
         verlet integrator while computing the trajectory using Hamiltonian
-        dynamics.
+        dynamics. If not specified, it will be set to 1.
+    :param float trajectory_length: Length of a MCMC trajectory. If not
+        specified, it will be set to ``step_size x num_steps``. In case
+        ``num_steps`` is not specified, it will be set to :math:`2\pi`.
     :param int num_steps: The number of discrete steps over which to simulate
         Hamiltonian dynamics. The state at the end of the trajectory is
-        returned as the proposal.
+        returned as the proposal. This value is always equal to
+        ``int(trajectory_length / step_size)``.
+    :param bool adapt_step_size: A flag to decide if we want to adapt step_size
```

Comment:
Can we internally set `self._adapt_step_size = True if step_size is None else False`, for NUTS and HMC? That way we will have one less argument to worry about, and avoid invalid configs (like `adapt_step_size=True, step_size=0.2`).